eZ/Publish/Core/Persistence/Legacy/User/Role/Gateway/ExceptionConversion.php

```diff
@@ -38,307 +38,160 @@ public function __construct(Gateway $innerGateway)
         $this->innerGateway = $innerGateway;
     }
 
-    /**
-     * Create new role.
-     *
-     * @param Role $role
-     *
-     * @return Role
-     */
-    public function createRole(Role $role)
+    public function createRole(Role $role): Role
     {
         try {
             return $this->innerGateway->createRole($role);
-        } catch (DBALException $e) {
-            throw new RuntimeException('Database error', 0, $e);
-        } catch (PDOException $e) {
-            throw new RuntimeException('Database error', 0, $e);
+        } catch (DBALException | PDOException $e) {
+            throw DatabaseException::wrap($e, DatabaseException::DEFAULT_MESSAGE, $e->getCode());
```

Comment:
Please leave only `throw DatabaseException::wrap($e)` right now for consistency and then we will improve `\eZ\Publish\Core\Base\Exceptions\DatabaseException::wrap` as a follow up story.